tests/ui/typeck/bad-index-due-to-nested.rs

```diff
@@ -0,0 +1,27 @@
+use std::hash::Hash;
```

Comment:
I would've marked this as `// run-rustfix` but I cannot due to overlapping suggestion spans. I'll fix that in a different PR.